scripts/cnode-helper-scripts/topologyUpdater.sh

```diff
@@ -51,24 +51,38 @@ shift $((OPTIND -1))
 
 [[ -z "${BATCH_AUTO_UPDATE}" ]] && BATCH_AUTO_UPDATE=N
 
+#######################################################
+# Version Check                                       #
+#######################################################
+clear
+
 if [[ ! -f "${PARENT}"/env ]]; then
   echo -e "\nCommon env file missing: ${PARENT}/env"
   echo -e "This is a mandatory prerequisite, please install with prereqs.sh or manually download from GitHub\n"
   exit 1
 fi
 
-# Check if update is available
-! checkUpdate env ${BATCH_AUTO_UPDATE} && exit 1
-! checkUpdate topologyUpdater.sh ${BATCH_AUTO_UPDATE} && exit 1
-
-# source common env variables in case it was updated and run in offline mode, even for TU_PUSH mode as this will be cought by failed EKG query
-# ignore exit code 0 and 2, any other exits script
-. "${PARENT}"/env offline
-case $? in
-  0) : ;; # ok
-  2) echo "continuing with topology update..." ;;
-  *) exit 1 ;;
-esac
+. "${PARENT}"/env offline &>/dev/null # ignore any errors, re-sourced later
+
+if [[ "${UPDATE_CHECK}" == "Y" ]] && [[ "${BATCH_AUTO_UPDATE}" == "N" ]]; then
+  echo "Checking for script updates..."
+  # Check availability of checkUpdate function
+  if [[ $(command -v checkUpdate) ]]; then
+    echo -e "\nCould not find checkUpdate function in env, make sure you're using official guild docos for installation!"
+    myExit 1
+  fi
+  # check for env update
+  ! checkUpdate env ${BATCH_AUTO_UPDATE} && myExit 1
+  ! checkUpdate topologyUpdater.sh ${BATCH_AUTO_UPDATE} && myExit 1
+  ! . "${PARENT}"/env && myExit 1 "ERROR: Topology Updater failed to load common env file\nPlease verify set values in 'User Variables' section in env file or log an issue on GitHub"
+  # source common env variables in case it was updated
+  . "${PARENT}"/env
```

Comment:
Double source:ing of env. Remove the first one and also make it offline, no need to source in online mode for TU.